types/ember__string/index.d.ts

```diff
@@ -7,8 +7,6 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.7
 
-export { htmlSafe, isHTMLSafe } from '@ember/template';
```

Comment:
I'm not sure removing this re-export was actually the right thing. See discussions in: #38571

Reply:
Gah, I had a vague memory there was something like that but when I went poking around I couldn’t find it.

Reply:
I can’t actually find *either* any public documentation of the `@ember/string` exports *or* deprecations thereof, but I believe the issue that spawned this PR was due to `@ember/string`’s incorrectly publishing types, so presumably they *are* public? This seems to be a bit of a mess. :grimacing:

Reply:
It *is* a mess. The[`@ember/string` package](https://github.com/emberjs/ember-string) is written in TS and does indeed ship types (that don't include `htmlSafe` and `isHTMLSafe` because they are not implemented there), but that package is an addon of string utilities *extracted* from Ember. `ember-source` also provides `@ember/string`, which *does* include `htmlSafe` and `isHTMLSafe` as demonstrated in [this twiddle](https://ember-twiddle.com/6ef7ad27426bc5cfaafd053f729dc539?openFiles=controllers.application%5C.js%2C) (they are added to `@ember/string` [here](https://github.com/emberjs/ember.js/blob/ce750104d0f4d2fac6dd7a9ee66a12a9411c3f5d/packages/ember/index.js#L654-L655) in `ember-source`, where `htmlSafe` is defined [here](https://github.com/emberjs/ember.js/blob/ce750104d0f4d2fac6dd7a9ee66a12a9411c3f5d/packages/%40ember/-internals/glimmer/lib/utils/string.ts#L61-L85) and `isHTMLSafe` is defined [here](https://github.com/emberjs/ember.js/blob/ce750104d0f4d2fac6dd7a9ee66a12a9411c3f5d/packages/%40ember/-internals/glimmer/lib/utils/string.ts#L87-L108)). The `@ember/string` *package* is not a dep of `ember-source`, but is *is* a dep of `ember-data` as of 3.24: https://github.com/emberjs/data/pull/7356. Twiddle only goes up to 3.18, so I tried the same thing I did there locally with a fresh 3.24 app and it still works (is functional) but fails to typecheck. Just downgrading to `@types/ember__string@3.16.1` does not work in this case because `ember-data@3.24` has installed the `@ember/string` package and TS is using it's types rather than `@types/ember__string`. If I remove `ember-data` (and thus the `@ember/string` package) with the downgraded `@types/ember__string`, it typechecks (and is functional). A solution that *does* work for both cases, however, is using module augmentation here instead of module declaration. I'll open a PR shortly for this.

Reply:
#50653

Reply:
Oh, wait found more info!: https://github.com/emberjs/rfc-tracking/issues/26
IMHO we should leave these in until *removed* (which I assume won't be until Ember 4.0).